syntax = "proto3";

message DashboardRequest {
  bytes requestJson = 1;
}

message Response {
  int32 code = 1;
  string msg = 2;
  bytes resultJson = 3;
}

message WeeklyCase {
  string date = 1;
  int32 count = 2;
}

message WorkSpaceData {
  int32 project_count = 1;
  int32 case_count = 2;
  int32 user_rank = 3;
  repeated WeeklyCase weekly_case = 4;
  int32 total_user = 5;
}

message WorkspaceResponse {
  int32 code = 1;
  string msg = 2;
  WorkSpaceData data = 3;
}

message Rank {
  int32 id = 1;
  int32 count = 2;
  int32 rank = 3;
}

message Data {

}

message Count {
  optional int32 project = 1;
  optional int32 testcase = 2;
  optional int32 testplan = 3;
  optional int32 user = 4;
}

message Report {
  int32 count = 1;
  optional int32 error = 2;
  optional int32 failed = 3;
  optional int32 rate = 4;
  optional int32 skip = 5;
  optional int32 success = 6;
  string date = 7;
}

message ReportData {
  string date = 1;
  int32 count = 2;
  optional int32 error = 3;
  optional int32 failed = 4;
  optional int32 rate = 5;
  optional int32 skip = 6;
  optional int32 success = 7;
  repeated Report data = 8;
}

message CountData {
  string date = 1;
  int32 project = 2;
  optional int32 testplan = 3;
  optional int32 user = 4;
  optional int32 testcase = 5;
}

message StatisticsData {
  Count count = 1;
  repeated Rank rank = 2;
  ReportData report = 3;
  repeated CountData data = 4;
}

message StatisticsResponse {
  int32 code = 1;
  string msg = 2;
  StatisticsData data = 3;
}

service dashboard {
  rpc statistics(DashboardRequest) returns (StatisticsResponse) {}
  rpc workspace(DashboardRequest) returns (WorkspaceResponse) {}
}